### PR TITLE
feat: expand subgraph object

### LINKED ISSUE #32: Add mappings.ts files to send to chatgpt

### LINKED ISSUE #34: support protocol forks in subgraphs.py
eg `dydx-governance` in `aave-governance/protocols/`

### PR DESCRIPTION
**Pull request by gosuto-inzasheru**
solves #34

### CODE REVIEW

**Comment by gosuto-inzasheru** [comment]
bonus: talk to gpt from cli using `python chat.py`

File: backend/backend/config.py [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Use env var to override
 """
 import os
+from dotenv import load_dotenv
+
+
+load_dotenv()
```
Comment (townie) [inline backend/backend/config.py:9]: This should happen automagically, but maybe not if running outside of flask app

File: backend/backend/services/graph/subgraphs.py [hunk 2]
```diff
@@ -9,63 +9,57 @@
...
+            in self.deployments["deployments"][f"{protocol}-{chain}"]["services"]
+        ):
+            self.service_type = "hosted-service"
+        try:
+            self.query_id = self.deployments["deployments"][f"{protocol}-{chain}"][
+                "services"
+            ][self.service_type]["query-id"]
+        except AttributeError:
+            # protocol does not have any deployments
+            raise NotImplementedError
```
Comment (townie) [inline backend/backend/services/graph/subgraphs.py:43]: nit, we should make sure we are catching the raise on the controller and returning a proper status code.

File: backend/backend/services/graph/subgraphs.py [hunk 3]
```diff
@@ -9,63 +9,57 @@
...
+        try:
+            self.template_file_location = f"subgraphs/subgraphs/{self.deployments['base']}/protocols/{self.protocol}/config/templates/{self.deployments['deployments'][f'{protocol}-{chain}']['files']['template']}"
+        except:
+            # no template file defined
```
Comment (townie) [inline backend/backend/services/graph/subgraphs.py]: We should still set this `self.template_file_location = None` for proper object typing

File: backend/backend/services/graph/subgraphs.py [hunk 4]
```diff
@@ -9,63 +9,57 @@
...
+            pass
+        if os.path.isfile(
+            f"subgraphs/subgraphs/{self.deployments['base']}/schema.graphql"
+        ):
+            self.schema_file_location = (
```
Comment (townie) [inline backend/backend/services/graph/subgraphs.py:52]: same nit, We should still set this `self.`schema_file_location = None` for proper object typing

File: backend/backend/services/openAI/service.py [hunk 5]
```diff
@@ -18,16 +18,22 @@ def generate_prompt(input):
 
 
 class OpenAIService:
-  def __init__(self, use_prompt=0):
-    openai.api_key = OPENAI_API_KEY
+    def __init__(self, use_prompt=0):
+        openai.api_key = OPENAI_API_KEY
```
Comment (townie) [inline backend/backend/services/openAI/service.py:22]: I been thinking about this, it's nice to have it here for using any script, but its nice to have it happen at app start up in the flask app also

File: backend/backend/services/openAI/service.py [hunk 6]
```diff
@@ -18,16 +18,22 @@ def generate_prompt(input):
...
-      )
-      openai_result = response.choices[0].text
-      print("==========openai response:==========\n", openai_result)
-      return openai_result
+    def request_gql_for_graph(self, input_query, subgraph, generic=False):
+        prompt = input_query if generic else generate_prompt(input_query)
+        response = openai.Completion.create(
+            model="text-davinci-003",
+            prompt=generate_prompt(input_query),
+            temperature=0.6,
```
Comment (townie) [inline backend/backend/services/openAI/service.py:29]: We should param these but outside of scope of pr

File: frontend/src/Components/CustomizedInputBase.js [hunk 7]
```diff
@@ -78,6 +78,7 @@ export function CustomizedInputBase(props) {
           label="SubGraph"
           onChange={handleChange}
         >
+          <MenuItem value={"aave-governance"}>aave-governance</MenuItem>
```
Comment (townie) [inline frontend/src/Components/CustomizedInputBase.js:81]: nice

**Review by townie — DISMISSED** [review]

File: backend/backend/config.py (hunk 1 above)
Comment (gosuto-inzasheru) [reply backend/backend/config.py:9]: ah i see. yea for debugging in sometimes run outside of flask

File: backend/backend/services/graph/subgraphs.py (hunk 3 above)
Comment (gosuto-inzasheru) [reply backend/backend/services/graph/subgraphs.py]: e41910facc691dd7c5c3236ebf599763cd691973

File: backend/backend/services/graph/subgraphs.py (hunk 2 above)
Comment (gosuto-inzasheru) [reply backend/backend/services/graph/subgraphs.py:43]: yes, lets do that as part of the whole error catching and sending to f/e part

Comment (townie) [reply backend/backend/services/graph/subgraphs.py:43]: 👍

**Review by townie — APPROVED** [review]

**Pull request merged**